sdk/src/types/block/address/bech32.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::str::FromStr;
+use std::hash::Hash;
+
+use crate::types::block::{address::Address, Error};
+
+/// An address and its network type.
+#[derive(Clone, Eq, PartialEq, Hash)]
+pub struct Bech32Address {
+    pub(crate) hrp: String,
+    pub(crate) inner: Address,
+}
+
+impl AsRef<Address> for Bech32Address {
+    fn as_ref(&self) -> &Address {
+        &self.inner
+    }
+}
+
+impl FromStr for Bech32Address {
+    type Err = Error;
+
+    fn from_str(address: &str) -> Result<Self, Self::Err> {
+        let (hrp, inner) = Address::try_from_bech32_with_hrp(address)?;
+        Ok(Self { hrp, inner })
+    }
+}
+
+impl Bech32Address {
+    /// Create a new address wrapper.
+    pub fn new(address: Address, hrp: String) -> Result<Self, Error> {
```

Comment:
https://github.com/rust-bitcoin/rust-bech32/issues/99

Reply:
Looks like they won't do it right now :/

Reply:
I put a TODO, let's merge and revisit when they make progress